api/src/test/java/org/openmrs/module/pihcore/status/StatusDataFunctionsTest.java

```diff
@@ -19,20 +19,30 @@ public void testVelocityExpressions() {
         Calendar cal = Calendar.getInstance();
         DateFormat df = new SimpleDateFormat("yyyy-MM-dd");
         String today = df.format(cal.getTime());
+
+        cal = Calendar.getInstance();
         cal.add(Calendar.YEAR, 3);
         dataValues.put("in3Years", cal.getTime());
-        cal.add(Calendar.YEAR, -3);
+
+        cal = Calendar.getInstance();
         cal.add(Calendar.DATE, 52);
         dataValues.put("in52Days", cal.getTime());
-        cal.add(Calendar.DATE, -52);
+
+        cal = Calendar.getInstance();
         cal.add(Calendar.DATE, -1);
         dataValues.put("yesterday", cal.getTime());
         String yesterday = df.format(cal.getTime());
```

Comment:
When run on leap day (Feb 29th) "yesterday" was getting set to Feb 27 2024 (not Feb 28th) because we were reusing a calendar instance, and so when jumping three years again from Feb 29, 2024 the calendar ends up on Feb 28.2027, and so when you subtract 3 years you end up on Feb 28, 2024, not Feb 29. 2024.

I changed these tests to instantiate a new calendar for each new element.